api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -69,13 +69,11 @@
      *
      * @param propertyName
      *             The configuration propertyName.
-     * @return The resolved property value as a String-Optional, which will bypass converters.
+     * @return The resolved property value as a String, which will bypass any converters.
      *
-     * @throws IllegalArgumentException
-     *             is thrown if the propertyName maps to an object that is not a
-     *             String.
+     * 
      */
-    Optional<String> getString(String propertyName);
+    String getRawString(String propertyName);
```

Comment:
IIRC, it is really meant for debugging purposes to see what is the final value over all config sources (even if multiple sources provide it), but before it goes into a converter. Since converters work with String, it's natural to return raw value as a String.

Maybe we should leave this method out for now, and only suggest that implementations provide vendor-specific means to access the final raw value, provided that it is only for debugging as I assume.